torch/csrc/jit/codegen/cuda/executor.cpp

```diff
@@ -329,44 +353,62 @@ uint64_t FusionExecutor::computeSharedMemory(
   return total;
 }
 
+uint64_t FusionExecutor::computeSharedMemory(
+    kir::ExpressionEvaluator& expr_eval,
+    const std::vector<const kir::Allocate*>& buffers,
+    const std::vector<glfdc::ExprEvaluator>& memoized,
```

Comment:
`Memoized` looks like logic on the `Int`'s instead of the actual allocation or shared mem computation. Does it make sense to wrap the memoized argument in `kir::ExpressionEvaluator`? Or maybe some other alternatives to having to provide a memoized argument to all functions that could benefit from memoization.

Reply:
I will have to think bit more about it. I've already did give it some thought, since it would be nice to somehow integrate it better with kir altoghter.